packages/ember-metal/lib/streams/stream.js

```diff
@@ -40,6 +41,8 @@ BasicStream.prototype = {
     this.dependencyHead = null;
     this.dependencyTail = null;
     this.observedProxy = null;
+    this.__ember_meta__ = null;
+    this[GUID_KEY] = null;
```

Comment:
The code in `guidFor` explicitly checks for `null` first and does `obj[GUID_KEY] = newGuid` (this is [done here](https://github.com/emberjs/ember.js/blob/master/packages/ember-metal/lib/utils.js#L240-L241)) before trying `obj.__defineNonEnumerable`.

I don't mind swapping this to `undefined` and adding `__defineNonEnumerable`, but ensuring the slot is available via `null` does fit with the existing system.

Please let me know which way you want me to go...